src/zos.cc

```diff
@@ -3104,10 +3104,16 @@ extern "C" void *__aligned_malloc(size_t size, size_t alignment) {
     return ptr;
   return nullptr;
 #else
+  if (size == 0)
```

Comment:
This might give different behaviour from malloc(0) or posix_memalign(&ptr, align, 0).

Reply:
According to [posix_memalign on z/OS](https://www.ibm.com/docs/en/zos/3.1.0?topic=functions-posix-memalign-reserve-aligned-storage-block): "If size was specified to 0, posix_memalign() returns zero and sets memptr to a null pointer."